wagtail/wagtailsearch/index.py

```diff
@@ -77,6 +78,34 @@ def get_indexed_instance(self):
         """
         return self
 
+    @classmethod
+    def _has_field(cls, name):
+        try:
+            cls._meta.get_field(name)
+            return True
+        except models.fields.FieldDoesNotExist:
+            return hasattr(cls, name)
+
+    @classmethod
+    def check(cls, **kwargs):
+        errors = super(Indexed, cls).check(**kwargs)
+        errors.extend(cls._check_fields_exist(**kwargs))
+        return errors
+
+    @classmethod
+    def _check_fields_exist(cls, **kwargs):
+        errors = []
+        for field in cls.get_search_fields():
+            message = '{name} in search_fields but not on {model}'
+            if not cls._has_field(field.field_name):
```

Comment:
This might be neater to implement on the ``BaseField`` class below as that class is responsible to doing the actual lookup. Could we move this to a new method on ``BaseField``? (eg ``BaseField.check(model)``)